app/scripts/metamask-controller.js

```diff
@@ -2413,11 +2413,18 @@ export default class MetamaskController extends EventEmitter {
   }
 
   triggerNetworkrequests() {
+    const filtredChainIds = this.#getAllAddedNetworks().filter(
+      (networkId) =>
+        this.preferencesController.state.incomingTransactionsPreferences[
+          networkId
+        ],
+    );
+
     this.txController.stopIncomingTransactionPolling();
 
-    this.txController.startIncomingTransactionPolling([
-      this.#getGlobalChainId(),
-    ]);
+    this.txController.stopIncomingTransactionPolling();
+
+    this.txController.startIncomingTransactionPolling(filtredChainIds);
```

Comment:
Nice!

Should we add a guard that we aren't starting polling if the `filteredChainIds.length === 0` ?